src/main/cpp/src/list_slice.hpp

```diff
@@ -23,27 +23,188 @@
 
 namespace spark_rapids_jni {
 
+/**
+ * @brief Slices each row of a lists column according to the requested `start` and `length`.
+ *
+ * The indices start at 1, or from the end if negative(can not be zero). If any index in start is
+ * outside [-n, n] for a row (where n is the number of elements in that row), the result for that
+ * row is an empty list.
+ *
+ * If length is zero, the result for that row is an empty list. If length exceeds the number of
+ * elements from the real start index(0 to n - 1) to the end of the row, the result for that row
+ * is the elements from the real start index to the end of the row.
```

Comment:
The element number of result list is less than sepcified length if there are not enough ements from the specifed start to the end of the target list.

Reply:
Require the length of the three parameter are equal.